stac_geoparquet/arrow/_batch.py

```diff
@@ -83,6 +87,9 @@ def from_dicts(
         # `ArrowInvalid: cannot mix list and non-list, non-null values`
         wkb_items = []
         for item in items:
+            if isinstance(item, pystac.Item):
+                item = item.to_dict()
```

Comment:
Re: https://github.com/stac-utils/pystac/issues/960, you probably want to ensure that you don't try to resolve the root link (this is an annoying **pystac** quirk):

```suggestion
                item = item.to_dict(tranform_hrefs=False)
```